modules/runtime/nuxeo-runtime-cluster/src/main/java/org/nuxeo/runtime/cluster/ClusterServiceImpl.java

```diff
@@ -67,7 +66,7 @@ public int getApplicationStartedOrder() {
 
     @Override
     public void start(ComponentContext context) {
-        ClusterNodeDescriptor descr = getDescriptor(XP_CONFIG, UNIQUE_DESCRIPTOR_ID);
+        ClusterNodeDescriptor descr = (ClusterNodeDescriptor) getRegistryContribution(XP_CONFIG).orElse(null);
```

Comment:
Is the cast needed?

Reply:
Yes: "Type mismatch: cannot convert from Object to ClusterNodeDescriptor"